### PR TITLE
Gmail enhancement fetch incident

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-2553

## Description
A few sentences describing the overall goals of the pull request's commits.

## Minimum version of Cortex XSOAR
- [X] 6.0.0

## Does it break backward compatibility?
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 1]
```diff
@@ -230,24 +231,79 @@ def create_base_time(internal_date_timestamp, header_date):
     return base_time
 
 
+def get_occurred_date(email_data: dict) -> Tuple[datetime, bool]:
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:234]: Please add a test

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 2]
```diff
@@ -230,24 +231,79 @@ def create_base_time(internal_date_timestamp, header_date):
...
+        timestamp_len = len(str(int(time.time())))
+        if len(str(internalDate)) > timestamp_len:
+            internalDate = (str(internalDate)[:timestamp_len])
+        return datetime.fromtimestamp(int(internalDate), tz=timezone.utc), True
+        # we didn't get a date from anywhere
+    demisto.info("Failed finding date from internal or headers. Using 'datetime.now()'")
+    return datetime.now(tz=timezone.utc), False
+
+
+def get_date_from_email_header(header: str) -> Optional[datetime]:
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:270]: Same, add test

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 3]
```diff
@@ -420,18 +486,18 @@ def mail_to_incident(msg, service, user_key):
             'name': attachment['Name'],
         })
     # date in the incident itself is set to GMT time, the correction to local time is done in Demisto
-    gmt_time = move_to_gmt(parsed_msg['Date'])
+    # gmt_time = move_to_gmt(parsed_msg['Date'])
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:489]: If we don't need this anymore - remove the comment

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 4]
```diff
@@ -2185,6 +2251,16 @@ def send_as_add_command():
     )
 
 
+def parse_date_isoformat_server(dt: str) -> datetime:
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2254]: This is the same as `get_date_isoformat_server` function, no?

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 5]
```diff
@@ -2314,50 +2390,90 @@ def forwarding_address_list_command() -> CommandResults:
 
 def fetch_incidents():
     params = demisto.params()
-    user_key = params.get('queryUserKey')
-    user_key = user_key if user_key else ADMIN_EMAIL
+    max_fetch = int(params.get('fetch_limit') or 100)
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2393]:
```suggestion
    max_fetch = int(params.get('fetch_limit') or 50)
```

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 6]
```diff
@@ -2314,50 +2390,90 @@ def forwarding_address_list_command() -> CommandResults:
 
 def fetch_incidents():
     params = demisto.params()
-    user_key = params.get('queryUserKey')
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2317]: Why did you remove this?

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 7]
```diff
@@ -2314,50 +2390,90 @@ def forwarding_address_list_command() -> CommandResults:
...
-    user_key = user_key if user_key else ADMIN_EMAIL
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2318]: And this?

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 8]
```diff
@@ -2314,50 +2390,90 @@ def forwarding_address_list_command() -> CommandResults:
...
+    max_fetch = int(params.get('fetch_limit') or 100)
+    user_key = 'me'
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2394]: The user_key always is `me`?

**Review by Shellyber — CHANGES_REQUESTED** [review]
Nice

File: Packs/Gmail/Integrations/Gmail/Gmail.py (hunk 4 above)
Comment (sapirshuker) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:2254]: No, parse_date_isoformat_server gets a string and returns a datetime in a specific format and time zone (used to set lastRun obj) and get_date_isoformat_server gets datetime and returns a parsed string (used to create an incident)

Comment (Shellyber) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:2254]: Let's add tests for both

File: Packs/Gmail/Integrations/Gmail/Gmail.py (hunk 6 above)
Comment (sapirshuker) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:2317]: added

**Review by Shellyber — APPROVED** [review]
Nice Work
